techdocs/docs/using-the-package/apis-&-components/protected-route.md

```diff
@@ -0,0 +1,120 @@
+# protectedRoute
+
+The `protectedRoute` function is middleware for your express routes that protects the endpoint so only logged in users can use it, and can be used to require the user to have a specific role to access the endpoint.
+
+## Import
+
+```JavaScript
+// ESModule Syntax (preferred)
+import { protectedRoute } from "@bcgov/citz-imb-sso-express";
+
+// CommonJS Syntax
+const { protectedRoute } = require('@bcgov/citz-imb-sso-express');
+```
+
+## Usage
+
+### Where Express App is Defined
+
+A basic example of using the `protectedRoute` middleware on the express app.
+
+```JavaScript
+import express, { Application } from 'express';
+import { protectedRoute } from '@bcgov/citz-imb-sso-express';
+
+// Define Express App.
+const app = express();
+
+// Initialize SSO.
+sso(app);
+
+app.use("/protected", protectedRoute(), protectedRouter); // Protected.
+app.use("/unprotected", unprotectedRouter); // Not protected.
+```
+
+### In a Router
+
+A basic example of using the `protectedRoute` middleware on an express router.
+
+```JavaScript
+import express from "express";
+const router = express.Router();
+import { protectedRoute } from '@bcgov/citz-imb-sso-express';
+
+router.get("/protected", protectedRoute(), protectedController()); // Protected.
+router.get("/unprotected", unprotectedController()); // Not protected.
+```
+
+!!! warning "Warning"
+    Protecting of routes with the same route string will not work. Each route string must be unique.  
+    As an example `router.get("/api", unprotectedController());` and `router.get("/api", protectedRoute(), protectedController());` will not work.
+
+
+### Requiring Roles for a Route
+
+You can require that a user has a role or roles in order to use a protected route by passing an array of strings (role names) to the `protectedRoute` middleware function.
+
+```Javascript
+// Users must have 'Member' role.
+app.use("/post", protectedRoute(['Member']), postRouter);
+
+// Users must have BOTH 'Member' and 'Commenter' roles.
+// requireAllRoles option is true by default.
+app.use("/comment", protectedRoute(['Member', 'Commenter']), commentRouter);
+
+// Users must have EITHER 'Member' or 'Verified' role.
+app.use("/vote", protectedRoute(['Member', 'Verified'], { requireAllRoles: false }), voteRouter);
+```
+
+!!! note "Note"
+    By default, all roles in the array will be required.  
+    If you wish to require only one of a list of roles, add the options parameter with `requireAllRoles` property set to `false`.
+
+!!! tip "Tip"
+    You can also check if the user has a role within the controller function by using the `hasRoles` function on the `SSOUser` object of `req.user`.
+
+## TypeScript Type
+
+```TypeScript
+(roles?: string[], options?: ProtectedRouteOptions) => RequestHandler;
+```
```

Comment:
do we want this here, or just a note/ tip to checkout the typescript type page?

Reply:
I see this page as being all about the function, where as the typescript types page is a collection of everything.

Reply:
my only worry is that if we update something then we have to hunt for all the references of it, instead of just updating the information in one place

Reply:
Then I think we get rid of the typescript types page

Reply:
This is more clear to the user

Reply:
But at the same time the TypeScript types page is going to have types not listed in a component doc
and if we were to link off to the TypeScript Types page they'd have to search through it to find the type they're looking for

Reply:
I dont see it being much work to update the docs when something changes because we can use the search features of vscode